addons/account/models/account_journal.py

```diff
@@ -693,10 +717,12 @@ def _fill_missing_values(self, vals, protected_codes=False):
             vals['refund_sequence'] = vals['type'] in ('sale', 'purchase')
 
         # === Fill missing alias name for sale / purchase, to force alias creation ===
-        if journal_type in {'sale', 'purchase'} and 'alias_name' not in vals:
-            vals['alias_name'] = self._alias_prepare_alias_name(
-                False, vals.get('name'), vals.get('code'), journal_type, company
-            )
+        if journal_type in {'sale', 'purchase'}:
+            if 'alias_name' not in vals:
+                vals['alias_name'] = self._alias_prepare_alias_name(
+                    False, vals.get('name'), vals.get('code'), journal_type, company
+                )
+            vals['alias_name'] = self._check_unique_alias(vals, company)
```

Comment:
why not trying to put the logic of the function `_check_unique_alias` inside the function `_alias_prepare_alias_name` ? This way the function `_alias_prepare_alias_name` always returns a unique name. It's just an idea so you might have arguments against. Also, this way, if set properly, the function `_sanitize_alias_name` would be called after, making sure that no special character are used

Reply:
Thank you for the thorough review!

Putting the logic inside `_alias_prepare_alias_name` was the initial idea, however that function is called in many other places, I wanted  `_check_unique_alias` to only be called on a journal creation that would create a new alias
(Looking at it now, I should change the name, as the function does not simply `check` for uniqueness :)